Changing between highlight and underline annotations

michalrentka opened the pull request:
Closes #985

mvasilak reviewed (changes requested):
* Tapping the "Convert to ..." cell, and this changing without actually changing the annotation seems unintuitive. Maybe a cell with 2 radio buttons instead? Or make it an action in the `...` menu of the annotation instead of a cell, so it can be executed immediatelly.
* The colors for the highlight and the underline shouls be adopted for the annotation type. Converting an underline to a highlight results in a slightly darker color.
* Converting an annotation in the sidebar results in a jumpy animation and closes the reader.

michalrentka commented:
> * Tapping the "Convert to ..." cell, and this changing without actually changing the annotation seems unintuitive. Maybe a cell with 2 radio buttons instead? Or make it an action in the `...` menu of the annotation instead of a cell, so it can be executed immediatelly.
Updated based on #zotero discussion, changed to `UISegmentedControl` as you suggested.

> * The colors for the highlight and the underline shouls be adopted for the annotation type. Converting an underline to a highlight results in a slightly darker color.
I saw this before but I can't reproduce now. Anyway I do update the color of annotation when the type changes, so that shouldn't be the issue.

> * Converting an annotation in the sidebar results in a jumpy animation and closes the reader.
Fixed.

Check 52183050b2b6e5481305950a07387071f03c93bc

mvasilak reviewed (approved):
Can the cell change be made with a reconfigure instead of deletion and insertion, for a smoother animation?

michalrentka commented:
> Can the cell change be made with a reconfigure instead of deletion and insertion, for a smoother animation?

I tried to mitigate this, but since we're actually deleting one annotation and creating a new one with new key and Realm doesn't provide a good way to simply ignore a specific notification tied to those actions, I don't have a good way to change it. 
I could temporarily store the old and new key, wait for those 2 notifications and then just ignore original actions and create a refresh based on that data. But in case something else comes inbetween (it shouldn't, but yeah I wouldn't want to risk it), it'll crash. I'm not sure whether it's worth the work.
